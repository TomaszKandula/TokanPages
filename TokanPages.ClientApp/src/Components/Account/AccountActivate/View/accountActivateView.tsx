import * as React from "react";
import { ViewProperties } from "../../../../Shared/Abstractions";
import { CustomCard, Icon } from "../../../../Shared/Components";
import { ExtendedViewProps } from "../accountActivate";

interface AccountActivateViewProps extends ViewProperties, ExtendedViewProps {
    shouldFallback: boolean;
    caption: string;
    text1: string;
    text2: string;
    fallback: {
        caption: string;
        text: string[];
    };
    hasProgress: boolean;
    hasError: boolean;
    hasSuccess: boolean;
}

const ProblemIcon = <Icon name="Alert" size={3} />;
const AlertIcon = <Icon name="AlertCircle" size={3} />;
const CheckIcon = <Icon name="Check" size={3} />;
const InfoIcon = <Icon name="Information" size={3} />;

export const AccountActivateView = (props: AccountActivateViewProps): React.ReactElement => {
    return (
        <section className={`section ${props.background ?? ""}`}>
            <div className="bulma-container bulma-is-max-desktop">
                <div className={!props.className ? "pt-0 pb-15" : props.className}>
                    {props.shouldFallback ? (
                        <CustomCard
                            isLoading={props.isLoading}
                            caption={props.fallback?.caption}
                            text={props.fallback?.text}
                            icon={ProblemIcon}
                            colour="warning"
                        />
                    ) : (
                        <CustomCard
                            isLoading={props.isLoading}
                            caption={props.caption}
                            text={[props.text1, props.text2]}
                            icon={props.hasError ? AlertIcon : props.hasSuccess ? CheckIcon : InfoIcon}
                            colour={props.hasError ? "error" : props.hasSuccess ? "success" : "info"}
                        />
                    )}
                </div>
            </div>
        </section>
    );
};
