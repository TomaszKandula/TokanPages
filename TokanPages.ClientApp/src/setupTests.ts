// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
