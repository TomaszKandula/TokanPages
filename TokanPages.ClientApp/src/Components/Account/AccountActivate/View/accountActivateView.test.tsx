import "../../../../setupTests";
import React from "react";
import { render } from "@testing-library/react";
import { AccountActivateView } from "./accountActivateView";

describe("test account group component: accountActivateView", () => {
    it("should render correctly '<AccountActivateView />' when content is loaded.", () => {
        const html = render(
            <AccountActivateView
                isLoading={false}
                shouldFallback={false}
                caption={"Account Activation"}
                text1={"Your account has been successfully activated!"}
                text2={"You can now sign in."}
                hasProgress={false}
                hasError={false}
                hasSuccess={true}
            />
        );

        expect(html).toMatchSnapshot();
    });
});
