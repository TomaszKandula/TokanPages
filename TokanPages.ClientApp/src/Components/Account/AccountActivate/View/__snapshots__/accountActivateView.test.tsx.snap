// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test account group component: accountActivateView should render correctly '<AccountActivateView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="section "
      >
        <div
          class="MuiContainer-root container MuiContainer-maxWidthLg"
        >
          <div
            class="pt-0 pb-15"
          >
            <div
              class="MuiPaper-root MuiCard-root card MuiPaper-elevation0 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root card-content"
              >
                <div
                  class="text-centre mb-25"
                >
                  <div
                    class="mt-15 mb-15"
                  >
                    <div
                      class="MuiTypography-root aa-caption MuiTypography-body1"
                    >
                      Account Activation
                    </div>
                  </div>
                  <div
                    class="mt-40 mb-15"
                  >
                    <span
                      class="MuiTypography-root aa-text1 MuiTypography-body1"
                    >
                      Your account has been successfully activated!
                    </span>
                  </div>
                  <div
                    class="mt-15 mb-40"
                  >
                    <span
                      class="MuiTypography-root aa-text2 MuiTypography-body1"
                    >
                      You can now sign in.
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="section "
    >
      <div
        class="MuiContainer-root container MuiContainer-maxWidthLg"
      >
        <div
          class="pt-0 pb-15"
        >
          <div
            class="MuiPaper-root MuiCard-root card MuiPaper-elevation0 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root card-content"
            >
              <div
                class="text-centre mb-25"
              >
                <div
                  class="mt-15 mb-15"
                >
                  <div
                    class="MuiTypography-root aa-caption MuiTypography-body1"
                  >
                    Account Activation
                  </div>
                </div>
                <div
                  class="mt-40 mb-15"
                >
                  <span
                    class="MuiTypography-root aa-text1 MuiTypography-body1"
                  >
                    Your account has been successfully activated!
                  </span>
                </div>
                <div
                  class="mt-15 mb-40"
                >
                  <span
                    class="MuiTypography-root aa-text2 MuiTypography-body1"
                  >
                    You can now sign in.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
